pex/pex.py

```diff
@@ -51,41 +51,48 @@ class IsolatedSysPath(object):
     @staticmethod
     def _expand_paths(*paths):
         # type: (*str) -> Iterator[str]
+        seen = set()
         for path in paths:
+            if path in seen:
+                continue
+            seen.add(path)
```

Comment:
Is this set just an optimisation, or is trying to be a guarantee that all output paths are unique?

If it's meant to be a guarantee, as written, it seems possible that this could be violated, e.g.:

- the `abspath` codepath: `paths == ["a", "/path/to/cwd/a"]` while running with `/path/to/cwd` and thus get a sequence like `["a", "/path/to/cwd/a", "/path/to/cwd/a"]` (where the first two correspond to the first input element, and the last corresponds to the second input element)
- the `realpath` codepath: `paths == ["/a", "/link-to-a"]` where `os.path.realpath("/link-to-a") == "/a"` 
- the combination of the two: AFAIK, `os.path.realpath` will absolutise a relative path, and thus `not isabs(path)` implies `os.path.realpath (realpath) != path` and both extra `yield`s will happen

Maybe this private function is not ever actually called with problematic args, though?

Reply:
The questions the code raised are enough here to justify simplifying at the expense of slightly more work. Fixed.